packages/runtime/runtime-utils/src/dataStoreHelpers.ts

```diff
@@ -44,31 +33,33 @@ export function exceptionToResponse(err: any): IResponse {
             mimeType: "text/plain",
             status: responseErr.code,
             value: responseErr.message,
-            stack: responseErr.stack ?? getStack(),
+            get stack() { return responseErr.stack; },
         };
     }
+
+    // Capture error objects, not stack itself, as stack retrieval is very expensive operation, so we delay it
+    const errWithStack = generateErrorWithStack();
+
     return {
         mimeType: "text/plain",
         status,
         value: `${err}`,
-        stack: getStack(),
+        get stack() { return ((err?.stack) as (string|undefined)) ?? errWithStack.stack; },
     };
 }
 
-export function responseToException(response: IResponse, request: IRequest) {
+export function responseToException(response: IResponse, request: IRequest): Error {
     const message = response.value;
-    const err = new Error(message);
-    const responseErr = err as any as IResponseException;
-    responseErr.errorFromRequestFluidObject = true;
-    responseErr.message = message;
-    responseErr.code = response.status;
-    if (response.stack !== undefined) {
-        try {
-            // not clear if all browsers allow overwriting stack
-            responseErr.stack = response.stack;
-        } catch (err2) {}
-    }
-    return err;
+    const errWithStack = generateErrorWithStack();
+    const responseErr: Error & IResponseException = {
```

Comment:
Consider if we can make this IFluidError?  Best to use a class that extends LoggingError.  It's not easy if a suitable class doesn't already exist, I have thoughts here but haven't had time.

Reply:
I'd prefer to move in small steps here. We can make such change later if desired